### PR TITLE
UX improvements to Python Storage and credentials instantiation

### PR DESCRIPTION
**Pull request by paraseba**

### CODE REVIEW

**Review by mpiannucci — COMMENTED** [review]
much improved, some nits

File: icechunk-python/python/icechunk/credentials.py [hunk 1]
```diff
@@ -1,13 +1,120 @@
...
+        S3StaticCredentials(
+            access_key_id=access_key_id,
+            secret_access_key=secret_access_key,
+            session_token=session_token,
+            expires_after=expires_after,
+        )
+    )
+
+
+def s3_dont_sign_credentials() -> S3Credentials.DontSign:
```
[the remark this answers is not in the document]
Comment (paraseba) [reply icechunk-python/python/icechunk/credentials.py:43]: this seems to be what S3 calls it, for example you pass `--no-sign-request` in the CLI. But I'll change it to anonymous if that's what people expect

[the remark this answers is not in the document]
Comment (mpiannucci) [reply icechunk-python/python/icechunk/credentials.py:43]: I will defer to others I was just making sure I understood

[the remark this answers is not in the document]
Comment (paraseba) [reply icechunk-python/python/icechunk/credentials.py:43]: Good enough for me, let's make it `anonymous` ... we can always add an alias later.

File: icechunk-python/src/config.rs [hunk 2]
```diff
@@ -356,15 +356,15 @@ impl PyStorage {
...
     #[staticmethod]
-    pub fn in_memory() -> PyResult<Self> {
+    pub fn new_in_memory() -> PyResult<Self> {
         let storage = icechunk::storage::new_in_memory_storage()
             .map_err(PyIcechunkStoreError::StorageError)?;
 
         Ok(PyStorage(storage))
     }
 
     #[staticmethod]
```
[the remark this answers is not in the document]
Comment (paraseba) [reply icechunk-python/src/config.rs]: done

File: icechunk-python/python/icechunk/credentials.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (paraseba) [reply icechunk-python/python/icechunk/credentials.py]: done

File: icechunk-python/python/icechunk/_icechunk_python.pyi [hunk 3]
```diff
@@ -265,16 +272,16 @@ class Storage:
     """
 
     @staticmethod
```
[the remark this answers is not in the document]
Comment (paraseba) [reply icechunk-python/python/icechunk/_icechunk_python.pyi]: done

**Review by mpiannucci — APPROVED** [review]

**Pull request merged**